resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ResourceRequestHelperSuite.scala

```diff
@@ -0,0 +1,255 @@
+    assert(resourceInfo === ResourceInformation(CUSTOM_RES_1, 123, "m"))
+
+    val resourceInfo2: ResourceInformation = ResourceRequestTestHelper
+      .getResourceInformationByName(resource, CUSTOM_RES_2)
+    assert(resourceInfo2 === ResourceInformation(CUSTOM_RES_2, 10, "G"))
+  }
+
+  test("empty SparkConf should be valid") {
+    val sparkConf = new SparkConf()
+    ResourceRequestHelper.validateResources(sparkConf)
+  }
+
+  test("just normal resources are defined") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(DRIVER_MEMORY.key, "3G")
+    sparkConf.set(DRIVER_CORES.key, "4")
+    sparkConf.set(EXECUTOR_MEMORY.key, "4G")
+    sparkConf.set(EXECUTOR_CORES.key, "2")
+
+    ResourceRequestHelper.validateResources(sparkConf)
+  }
+
+  test("Memory defined with new config for executor") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(NEW_CONFIG_EXECUTOR_MEMORY, "30G")
+
+    val thrown = intercept[SparkException] {
+      ResourceRequestHelper.validateResources(sparkConf)
+    }
+    thrown.getMessage should include (NEW_CONFIG_EXECUTOR_MEMORY)
+  }
+
+  test("Cores defined with new config for executor") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(NEW_CONFIG_EXECUTOR_CORES, "5")
+
+    val thrown = intercept[SparkException] {
+      ResourceRequestHelper.validateResources(sparkConf)
+    }
+    thrown.getMessage should include (NEW_CONFIG_EXECUTOR_CORES)
+  }
+
+  test("Memory defined with new config, client mode") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(NEW_CONFIG_AM_MEMORY, "1G")
+
+    val thrown = intercept[SparkException] {
+      ResourceRequestHelper.validateResources(sparkConf)
+    }
+    thrown.getMessage should include (NEW_CONFIG_AM_MEMORY)
+  }
+
+  test("Memory defined with new config for driver, cluster mode") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(NEW_CONFIG_DRIVER_MEMORY, "1G")
+
+    val thrown = intercept[SparkException] {
+      ResourceRequestHelper.validateResources(sparkConf)
+    }
+    thrown.getMessage should include (NEW_CONFIG_DRIVER_MEMORY)
+  }
+
+  test("Cores defined with new config, client mode") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(DRIVER_MEMORY.key, "2G")
+    sparkConf.set(DRIVER_CORES.key, "4")
+    sparkConf.set(EXECUTOR_MEMORY.key, "4G")
+    sparkConf.set(AM_CORES.key, "2")
+    sparkConf.set(NEW_CONFIG_AM_CORES, "3")
+
+    val thrown = intercept[SparkException] {
+      ResourceRequestHelper.validateResources(sparkConf)
+    }
+    thrown.getMessage should include (NEW_CONFIG_AM_CORES)
+  }
+
+  test("Cores defined with new config for driver, cluster mode") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(NEW_CONFIG_DRIVER_CORES, "1G")
+
+    val thrown = intercept[SparkException] {
+      ResourceRequestHelper.validateResources(sparkConf)
+    }
+    thrown.getMessage should include (NEW_CONFIG_DRIVER_CORES)
+  }
+
+  test("various duplicated definitions") {
```

Comment:
This doesn't seem to be testing anything that the other tests aren't already.

Reply:
I would still keep this since it tests whether the message to the user contains all the invalid properties, not just an individual invalid property. This is not being tested in the other methods so I think this test still holds a value.